src/components/sidecar/get-truncated-title.test.ts

```diff
@@ -0,0 +1,45 @@
+import getTruncatedTitle from './get-truncated-title'
+
+describe('getTruncatedTitle', () => {
+  it('leaves a short title unmodified', () => {
+    const title = 'Hello world'
+    expect(getTruncatedTitle(title)).toBe(title)
+  })
+
+  it('does not add ... to a title of exact limit length', () => {
+    const title = 'contrib Tools and Helpers'
+    const expected = 'contrib Tools and...'
+    expect(getTruncatedTitle(title)).toBe(expected)
+  })
+
+  /* @TODO
```

Comment:
This is a bit long-winded, but I think we should try out a CSS solution to truncation - I think it might be less brittle (ie, adapt to available pixel space rather than requiring specific max character counts per use case), and might also give us some nice polish (eg splitting words at "natural" locations, since browser use the same logic for truncation as they do for hyphenation, IIRC).

> This is assuming we haven't already taken a shot at this, if we have, pls ignore! 🙇

Reply:
AFAIK it's not possible to truncate text on full words with CSS. Have you done it or seen it done before?

Reply:
@ashleemboyer Ah, great point, I haven't seen truncation on full words before in CSS. The intent here was mainly just making things less brittle - for example if the sidecar changes width, or if the text size changes due to user settings.

I think there might be small benefit to truncating not just on full words as well - eg if a heading contains a long word like "infrastructure", we can show the first part of the word:

![Frame 2323](https://user-images.githubusercontent.com/4624598/147026472-f0bab71a-64ee-403a-8b25-87ce784a73cc.png)

That being said, there are definitely downsides to the CSS approach as well, so probably should have emphasized "try out". One downside with the CSS approach that I've noticed is that the current active states have a font-weight shift, and this can cause the truncation to vary in order to prevent line wrapping....

![CleanShot 2021-12-21 at 21 51 12](https://user-images.githubusercontent.com/4624598/147027295-484d796f-55b7-426f-bb6b-2c3aead881df.gif)

But, this font-weight shift can also cause issues with the custom truncation approach as well - this is a bit of a convoluted example, but the 22-character title `Wow Waypoint wwwwwwwww` is a decent example of both how character counts can be a bit brittle since letter width varies, and on the potential jank caused by `font-weight` shifting (which is maybe even worse since it results in line wrapping, but MUCH less likely to happen so perhaps not as relevant):

![CleanShot 2021-12-21 at 22 18 13](https://user-images.githubusercontent.com/4624598/147030406-7a848f3f-ea42-4703-984f-90906f658748.gif)

All that being said, depending on folks opinions, the mid-word truncation might be a downside as well, maybe enough to outweigh potential brittleness.

Reply:
☝️ Tested out the CSS approach in https://github.com/hashicorp/dev-portal/pull/56, let me know what you think!

Reply:
I think you bring up great points on both sides and I think we need to have a discussion between engineering and design to decide how to move forward. We consciously decided to go with truncating text on full words with JavaScript for our first preview, so I think we need to table the discussion until after then since I know design is a little pressed for time and we already made a short term decision on this. I think if we're lucky we might also get some feedback on this from our January 5th previewers and hopefully that can also help us make an informed decision! Can we hold on to the #56 experiment to use as an artifact in the larger discussion?

Reply:
☝️ Sounds great - super on board with tabling the discussion. Will leave #56 up and will leave this PR as-is with the Javascript word-based truncation 💯